src/test/java/com/google/sps/servlets/SearchServletTest.java

```diff
@@ -0,0 +1,216 @@
+  public void GetCourses_AllParamsSet() {
+
+    request = createRequest(request, "CS 105", "Smith", "Spring 2020", "1,2,3");
+    List<Course> expectedCourses = new ArrayList<>();
+
+    addCourseEntity("CS 105", "Smith", "Spring 2020", 1);
+    expectedCourses.add(new Course("CS 105", "Smith", Long.valueOf(1), "Spring 2020"));
+
+    addCourseEntity("CS 105", "Smith", "Spring 2020", 2);
+    expectedCourses.add(new Course("CS 105", "Smith", Long.valueOf(2), "Spring 2020"));
+
+    addCourseEntity("CS 106", "Smith", "Spring 2020", 3);
+
+    List<Filter> filters = searchObject.getFilters(request); // Tested in makeFiltersAllParams.
+    List<Entity> results = searchObject.getResults(filters); // Tested in getResultsAllFilters.
+    List<Course> courses = searchObject.getCourses(results);
+
+    assertEqualsCourseArrays(courses, expectedCourses);
+  }
+
+  @Test
+  /* Ensure the whole doGet process works as expected given no search filters. */
+  public void GetCourses_NoParamsSet() {
+
+    request = createRequest(request, "", "", "select", "");
+    List<Course> expectedCourses = new ArrayList<>();
+
+    addCourseEntity("CS 105", "Smith", "Spring 2020", 1);
+    expectedCourses.add(new Course("CS 105", "Smith", Long.valueOf(1), "Spring 2020"));
+
+    addCourseEntity("CS 105", "Smith", "Spring 2020", 2);
+    expectedCourses.add(new Course("CS 105", "Smith", Long.valueOf(2), "Spring 2020"));
+
+    addCourseEntity("CS 106", "Smith", "Spring 2020", 3);
+    expectedCourses.add(new Course("CS 106", "Smith", Long.valueOf(3), "Spring 2020"));
+
+    List<Filter> filters = searchObject.getFilters(request); // Tested in makeFiltersAllParams.
+    List<Entity> results = searchObject.getResults(filters); // Tested in getResultsAllFilters.
+    List<Course> courses = searchObject.getCourses(results);
+
+    assertEqualsCourseArrays(courses, expectedCourses);
+  }
+
+  private void addCourseEntityWithList(
+      String name, String professor, String term, int units, List<Entity> entities, boolean toAdd) {
+    DatastoreService db = DatastoreServiceFactory.getDatastoreService();
+    Entity entity = new Entity("Course");
+    entity.setProperty("name", name);
+    entity.setProperty("professor", professor);
+    entity.setProperty("units", units);
+    entity.setProperty("term", term);
+    db.put(entity);
+    if (toAdd) {
+      entities.add(entity);
+    }
+  }
+
+  private void addCourseEntity(String name, String professor, String term, int units) {
+    DatastoreService db = DatastoreServiceFactory.getDatastoreService();
+    Entity entity = new Entity("Course");
+    entity.setProperty("name", name);
+    entity.setProperty("professor", professor);
+    entity.setProperty("units", units);
+    entity.setProperty("term", term);
+    db.put(entity);
+  }
+
+  private HttpServletRequest createRequest(
+      HttpServletRequest request, String name, String professor, String term, String units) {
```

Comment:
Nice use of a helper function! One critique though: when you call this function you end up entering a sequence of string params. This is a place where it would be really easy for someone to mix up the param order and malform the function call. One solution to this is to use an object so the params are mapped. A possible workaround is to comment in the function call what the params are (since it's not obvious):
```request = createRequest(request, /** name */ "", /** professor */ "", /** term */ "select", /** units */ "");```. Since this is a test I don't think this needs to be a big deal but I think using this workaround at least would be a good idea.